### ISSUE TITLE
ASPPA Site Access Problem

### ISSUE THREAD

**Issue by ghost** [Tizra/Tizra-Customer-Tracker#47]
I have a user that can access http://elibrary.asppa.org/registeruser?registrationStep=claimAccount&emailAddress=rhood@omnipress.com and setup a password just fine but then gets an error from IE when they try to go to http://elibrary.asppa.org.

Error message is: 
Internet Explorer cannot display the webpage
What you can try:
Diagnose Connection Problems
More Information

When they change the address to https://elibrary.asppa.org they get a message about a problem with the websites sercurity certificate but they are able to access the site by clicking "Continue to this website (not recommended)"

I can't seem to problem solve this and David F is out of the office.
Have you ever had anything like this happen before?

**Comment by ghost** [comment]
I just found out the Chrome gives the following error:

This webpage is not available
The webpage at http://elibrary.asppa.org/ might be temporarily down or it may have move permanently to a new web address.

Error 321 (net::ERR_INVALID_CHUNKED_ENCODING): Unknown error.

**Comment by daviddurand** [comment]
This is pretty odd, but seems like it may be related to something at their location that is playing games -- at least, I was able to register the rhood account, and access the site using the URL you sent.

https access to the custom URLs is not something we normally use (because we have a certificate for secure.agilepdf.com, and not for each client domain). The fact that that works makes it look (at least on the surface) like there might be some filtering going on --- except that the first set of accesses is working, which is definitely odd.

**Comment by daviddurand** [comment]
I just had one thought: I saw some log errors where someone was requesting access to a server with a final "." on the host.

Could the URL be getting an extra period that is causing trouble? This could happen in the email, perhaps -- or in the formatted text version of the email, if the user is cutting and pasting the URL.

**Comment by ghost** [comment]
I avoided putting punctuation around the URL.
They got http://elibrary.asppa.org/ from me, typically not in a sentence.

could it have something to do with trying to access files with http: in front of the address? I know you had mentioned something about removing that. Not sure if that would cause this, but the registration page doesn't have much customization on it and works, but everything else is customized and doesn't.

**Comment by daviddurand** [comment]
Got it. The message about the invalid chunked encoding is the key. They have a proxy server that does not like the use of chunked HTTP encoding.

We have been working a problem like this for NIH on the Amia site. Originally we would sometimes set chunked encoding on HTTP 1.0 requests, (which is an error), but we have stopped doing this. We will need to get some clear information about the proxy server configuration at their site, or at the least HTTP header information on what their browser is seeing, and perhaps schedule some interactive time to work out what the problem is.

So to be clear, you are not seeing this problem yourself, but reporting what the user is seeing?

The other HTTP/HTTPS issues you were asking about have to do with the fact that secure pages (delivered by https) cannot contain or refer to anything delivered by HTTP without causing browser security warnings -- that's because a malicious user could fake any HTTP content to be whatever they want. That does not seem to be the issue here.

**Comment by ghost** [comment]
I'm reporting what the customer is seeing, but I'm not seeing.

I used my registration address because I didn't want to keep resetting the customers account while I tried to piece together what was going on.

When might you be available to schedule some interactive time, or is that only if they can't tell you about the proxy server. Also, do you have some specific questions that I can send along to them to help you pinpoint the issue?

Thank you

**Comment by daviddurand** [comment]
Well, if they can tell me something about the proxy that helps me to narrow it down.

With Chrome they could gather a lot of useful information from the inspector. If they right click and "inspect element" then they can go to the network tab and we can see all the requests and what headers went back and forth -- that doesn't tell us what is happening between their proxy server and our server -- which is much harder because we can't easily log every access because of the volume of requests -- but it does give us a lot of information.

If we could do screen sharing at the same time, I could even control their browser to gather the information -- or at least see exactly what is happening.

I'm also investigating some other things I might do.

**Comment by daviddurand** [comment]
Are there any updates on this? We got some information, but we haven't had any followup in a while. I'd like to close this if we can find the source of the problem, or perhaps, find out that the changes that I've made so far actually did solve the problem... I don't think that they have, but it's been quite some time since we heard from anyone. I did not find any weirdness in my checks, but without more information about what to look at, I could easily be missing something.
